### ISSUE TITLE
Inconsistent issue reporting.

### ISSUE THREAD

**Issue by pvande** [jdkato/txtlint#502]
Installed via Homebrew on macOS.

Given the following input:

``` html
<html>
<body>
  <p>This paragraph has has a duplication.</p>
  <p>This paragraph has <b>has</b> a duplication.</p>
  <p>
    This paragraph has has a
    duplication on multiple lines.
  </p>

  <p>This paragraph has <b>has</b> an <font>obsolete</font> tag</p>
  <p>
    This paragraph has <b>has</b> a
    duplication on multiple lines.
  </p>
  <p>
    This paragraph has
    has a duplication
    across lines.
  </p>
</body>
</html>
```

I get the following output:

```
 sample.html
 3:21  error  'has' is repeated!  Vale.Repetition 
 4:1   error  'has' is repeated!  Vale.Repetition 
 6:20  error  'has' is repeated!  Vale.Repetition 

✖ 3 errors, 0 warnings and 0 suggestions in 1 file.
```

Despite the example containing _six_ instances of the same error, only three are actually being reported. Additionally, the second report (`<p>… has <b>has</b> …</p>`) is described as covering the entire line, rather than more useful description of the error location.

**Comment by adamchainz** [comment]
I have encounted what I think is a related issue. Take this example:

```html
<p>This is good. But this is bad.</p>
<p>This is good.
But this is bad.</p>
```

The second one triggers `proselint.But`: “Do not start a paragraph with a 'but'.” - *BUT* it's within the same paragraph!

```
$ vale example.html

 example.html
 3:1  error  Do not start a paragraph with   proselint.But
             a 'but'.

✖ 1 error, 0 warnings and 0 suggestions in 1 file.
```

I'm triggering this because I write my reStructuredText with one sentence per line, which `rst2html` copies into its output.
